ui/pages/unlock-page/unlock-page.component.js

```diff
@@ -64,6 +68,18 @@ export default class UnlockPage extends Component {
     }
   }
 
+  componentDidUpdate(prevProps) {
+    const { isUnlocked, history } = this.props;
+
+    if (
+      !prevProps.isUnlocked &&
+      isUnlocked &&
+      getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN
```

Comment:
@Teng-Web3 Do we even need the environment check?  What happens if this isn't here?

Reply:
I ask because removing the environment check may also fix: https://github.com/MetaMask/metamask-extension/issues/30187